src/shared_state.rs

```diff
@@ -161,11 +190,24 @@ impl Server {
         &mut self,
         options: &FinalizedOptions,
         addr: SocketAddr,
-        op: &str,
+        mut connect_op: Connect<'_>,
+        nkey: Option<&KeyPair>,
     ) -> io::Result<(Reader, Writer, ServerInfo)> {
         let mut stream = TcpStream::connect(&addr)?;
         let info = crate::parser::expect_info(&mut stream)?;
 
+        if !info.nonce.is_empty() {
+            let nkey = nkey.unwrap();
+            let sig = nkey.sign(info.nonce.as_bytes()).unwrap();
+            let sig = base64_url::encode(&sig);
+            connect_op.sig = Some(sig);
```

Comment:
Try to wipe memory as best as we can once the CONNECT is sent.

Reply:
Implemented buffer & string zeroing via the new `SecureVec` & `SecureString` structs that scramble their data on Drop